packages/logging_cloudwatch/aws_logging_cloudwatch/lib/src/cloudwatch_logger_plugin.dart

```diff
@@ -304,10 +312,58 @@ class CloudWatchLoggerPlugin extends AWSLoggerPlugin
     if (!_enabled) {
       return false;
     }
+
     final loggingConstraint = _getLoggingConstraint();
+
+    if (loggingConstraint.userLogLevel.containsKey(_userId)) {
+      final userLevel = loggingConstraint.userLogLevel[_userId]!;
+      if (_checkLogLevel(userLevel.categoryLogLevel, logEntry)) {
+        return true;
+      }
+      return logEntry.level >= userLevel.defaultLogLevel;
+    }
+    final isLoggableByCategory =
+        _checkLogLevel(loggingConstraint.categoryLogLevel, logEntry);
+    if (isLoggableByCategory) {
+      return true;
+    } else if (loggingConstraint.categoryLogLevel.isNotEmpty) {
+      return false;
+    }
     return logEntry.level >= loggingConstraint.defaultLogLevel;
   }
 
+  bool _checkLogLevel(
+    Map<String, LogLevel> categoryLogLevel,
+    LogEntry logEntry,
+  ) {
+    for (final entry in categoryLogLevel.entries) {
+      if (logEntry.loggerName.toLowerCase().contains(entry.key.toLowerCase())) {
+        return logEntry.level >= entry.value;
+      }
+    }
+    return false;
+  }
+
+  /// Test the _isLoggable method.
+  @protected
+  @visibleForTesting
+  bool testIsLoggable(LogEntry logEntry) => _isLoggable(logEntry);
+
+  /// Get user id for testing.
+  @protected
+  @visibleForTesting
+  String? get testUserId => _userId;
+
+  /// Set user id for testing.
+  @protected
+  @visibleForTesting
+  set testUserId(String? userId) => _userId = userId;
+
+  /// Get logging constraint for testing.
+  @protected
+  @visibleForTesting
+  LoggingConstraints get testLoggingConstraint => _getLoggingConstraint();
+
```

Comment:
we should not need to add these for testing. 
unit tests should not test private methods directly. private methods are implementation detail. instead unit test should test public apis. in this case the  public api is `handleLogEntry`. if the `handleLogEntry` wroks as expected then the implementation is corect.